### PR TITLE
Use mlflow and ubuntu versions in image tags

### PR DESCRIPTION
**Pull request by misohu**
Load Ubuntu version from ARGs
Use inputs in CI
Create manual trigger for CI.

### CODE REVIEW

File: .github/workflows/build_and_publish_call.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
+name: Build and Publish
+
+on:
+  workflow_dispatch:
+    inputs:
+      ubuntu-version:
+        description: "Ubuntu version for the base image"
+        default: "22.04"
```
Comment (beliaev-maksim) [inline .github/workflows/build_and_publish_call.yaml:8]: did not we consider 20.04 to use?

File: .github/workflows/build_and_publish_call.yaml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+      
+      - name: Build Docker image
+        run: |
+          docker build -t ${{ vars.DOCKER_HUB_CHARMED_IMAGE }}:${{ github.sha }} --build-arg VERSION=${{ inputs.mlflow-version }} --build-arg UBUNTU_VERSION=${{ inputs.ubuntu-version }} .
```
Comment (beliaev-maksim) [inline .github/workflows/build_and_publish_call.yaml:68]: should not we build multi arch ?

File: .github/workflows/on_push_pull_request_schedule.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  schedule:
+    - cron: "0 14 * * 1"
+
+jobs:
+  build_and_publish:
+    name: Build and Publish the image
+    uses: ./.github/workflows/build_and_publish_call.yaml
+    secrets:
+      DOCKER_HUB_CHARMED_USERNAME: ${{ secrets.DOCKER_HUB_CHARMED_USERNAME }}
+      DOCKER_HUB_CHARMED_ACCESS_TOKEN: ${{ secrets.DOCKER_HUB_CHARMED_ACCESS_TOKEN }}
```
Comment (beliaev-maksim) [inline .github/workflows/on_push_pull_request_schedule.yaml:16]: new line is missing

File: .github/workflows/build_and_publish_call.yaml (hunk 1 above)
Comment (misohu) [reply .github/workflows/build_and_publish_call.yaml:8]: 20.04 does not come with python 3.10 I would need to add ppa death snakes and install it from there .... 20.04 will install 3.8 but would have to adjust Dockerfile

File: .github/workflows/build_and_publish_call.yaml (hunk 2 above)
Comment (misohu) [reply .github/workflows/build_and_publish_call.yaml:68]: Dont understand ... like 22.04 and 20.04 ?

File: .github/workflows/build_and_publish_call.yaml (hunk 1 above)
Comment (beliaev-maksim) [reply .github/workflows/build_and_publish_call.yaml:8]: agreed

File: .github/workflows/build_and_publish_call.yaml [hunk 4]
```diff
@@ -1,11 +1,45 @@
...
+        required: true
+      DOCKER_HUB_CHARMED_ACCESS_TOKEN:
+        required: true
+  workflow_call:
+    inputs:
+      ubuntu-version:
+        description: "Ubuntu version for the base image"
+        default: "22.04"
+        type: string
+        required: false
```
Comment (beliaev-maksim) [inline .github/workflows/build_and_publish_call.yaml:27]:
```suggestion
```

File: .github/workflows/build_and_publish_call.yaml (hunk 1 above)
Comment (misohu) [reply .github/workflows/build_and_publish_call.yaml]: removed

**Review by beliaev-maksim — APPROVED** [review]

**Pull request merged**